### PR TITLE
bau: Add error logging for redis

### PR DESCRIPTION
**Pull request by merlinc**
### What changed

Add error logging to the Redis client

### Why did it change

Redis errors are thrown outside of the request lifecycle, so need to be specifically added.

### Environment variables

- [x] No environment variables were added or changed

### Other considerations

- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks

### CODE REVIEW

File: app/session.ts [hunk 1]
```diff
@@ -26,6 +26,10 @@ import connectRedis from "connect-redis";
 import session from "express-session";
 import redis, { ClientOpts } from "redis";
 
+import Logger, { getLogLabel } from "./utils/logger";
```
Comment (merlinc) [inline app/session.ts:29]: Normally I'd expect loggers to be supplied as a singleton (Ugh!).

That's not how it is setup here, so for now, I've included the logger as it appears in the `server.js`. It doesn't seem to be doing anything particularly complicated, but if we have weird logging issues, we should pass it in as a parameter to this module.

File: app/session.ts [hunk 2]
```diff
@@ -58,6 +62,13 @@ export const setupSession = (): express.RequestHandler => {
   }
 
   const redisClient = getRedisClient();
+  redisClient.on("error", (error: Error) => {
+    logger.error(
+      `Unable to start server because of ${error.message}`,
```
Comment (merlinc) [inline app/session.ts]:
```suggestion
      `Redis client connection error - ${error.message}`,
```

File: app/session.ts (hunk 1 above)
Comment (martin-bucinskas) [reply app/session.ts:29]: Yeah, seems a bit weird that the logger is not a singleton. Seems its being set in req.session?
May be a future refactor, don't depend logger being instanstiated..

**Review by martin-bucinskas — APPROVED** [review]
LGTM

**Pull request merged**